File: 34-find-first-and-last-position-of-element-in-sorted-array/find-first-and-last-position-of-element-in-sorted-array.py
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:

        def leftBinarySearch(array, val):
            i = 0
            j = len(array) - 1
            index = -1

            while (i <= j):
                mid = (i + j) // 2

                if (array[mid] == val):
                    index = mid
                    j = mid - 1
                elif (val < array[mid]):
                    j = mid - 1
                else: 
                    i = mid + 1

            return index

        def rightBinarySearch(array, val):
            i = 0
            j = len(array) - 1
            index = -1

            while (i <= j):
                mid = (i + j) // 2

                if (array[mid] == val):
                    index = mid
                    i = mid + 1
                elif (val < array[mid]):
                    j = mid - 1
                else: 
                    i = mid + 1

            return index

        left = leftBinarySearch(nums, target)
        right = rightBinarySearch(nums, target)
        

        return [left, right]

File: 34-find-first-and-last-position-of-element-in-sorted-array/find-first-and-last-position-of-element-in-sorted-array.js
var searchRange = function(nums, target) {
    let first = leftBS(nums, target)
    let last = rightBS(nums, target)

    function leftBS(arr, val) {
        let i = 0
        let j = arr.length - 1
        let output = -1

        while (i <= j) {
            let mid = Math.floor((i + j) / 2)
            if (arr[mid] == val) {
                output = mid
                j = mid -1
            }
            else if (arr[mid] > val) j = mid - 1
            else i = mid + 1
        }

        return output
    }

    function rightBS(arr, val) {
        let i = 0
        let j = arr.length - 1
        output = -1

        while (i <= j) {
            let mid = Math.floor((i + j) / 2)
            if (arr[mid] == val) {
                output = mid
                i = mid + 1
            }
            else if (arr[mid] < val) i = mid + 1
            else j = mid - 1
        }

        return output
    }

    return [first, last]
};